### ISSUE TITLE
Fetch article was done twice with two formats param

### ISSUE THREAD

**Issue by automactic** [openzim/mwoffliner#695]
when running mwoffliner with `--format=nopic --format=novid`. The content is fetched twice. It would be great to only fetch the content once and generate two files.

For example, I was running mowffliner with `ab.wikipedia.org`

In the middle of the progress:
<img width="1557" alt="Screen Shot 2019-04-20 at 3 46 48 PM" src="https://user-images.githubusercontent.com/8294252/56462054-6ef99d80-638a-11e9-94cf-b4b13b25052b.png">

In the end of the progress.
<img width="1557" alt="Screen Shot 2019-04-20 at 4 36 12 PM" src="https://user-images.githubusercontent.com/8294252/56462069-92bce380-638a-11e9-9f8d-d0a5937635f1.png">

**Comment by kelson42** [comment]
@automactic This is exactly what happen if you use a cache... so why do we need something more?

**Comment by automactic** [comment]
Maybe I am understanding "cache" wrong.

If use cache, scrape twice. How many times is the scrape if I don't use cache?

**Comment by kelson42** [comment]
@automactic for `--format=nopic --format=novid`, only one time.

**Comment by automactic** [comment]
If use cache, I expect the logic to be:

```
def get_resources(url):
    if use_cache and already_cached:
        return cached_resource
    
    resource = go_to_the_internet_to_download_stuff(url)
    if use_cache:
        save_resource_as_cache(url, resource)
    return resource
```

**Comment by kelson42** [comment]
@automactic this is how it works, and otherwise this is a bug

**Comment by ISNIT0** [comment]
@kelson42 If this is how it should work, we need to store all the images and articles on disk again... So we lost the benefit of introducing node-libzim

**Comment by kelson42** [comment]
@ISNIT0  We don't loose all the benefit of libzim-node by using `--useCache`. Please open a separate ticket if `--useCache` does not work like @automactic has described, and we have here a regression too (because this is how I have implemented it).

All that said, I understand that if we could write many ZIM files at the same time smartly, ie. without having to have `--useCache` to avoid multiple HTTP requests to the same online resources, that would be better... but not sure we really need this?!

**Comment by ISNIT0** [comment]
Okay, I will create a new ticket, I've now implemented the caching and am testing it with a few different configurations.

I think the ideal situation really is to do things in parallel, otherwise we still have lots of I/O. This would be a relatively big task though.

**Comment by kelson42** [comment]
@ISNIT0 OK, please open quickly this ticket about the --useCache regression. I'm curious to see what goes wrong here. Considering the parallel writting, I will open a separate ticket which is a bit more clear about what is needed.

**Comment by kelson42** [comment]
As the two topics which has been discussed have been respectively report in #700 and #701, we can close this ticket.

**Comment by automactic** [comment]
To be honest, I don't really care about how it is implemented. (sorry!) Just let me know if I should use `useCache` if I want to only download stuff once.

**Comment by kelson42** [comment]
@automactic with 1.8.5 it should

**Comment by automactic** [comment]
> with 1.8.5 it should

Please clarify. 

What does it refer to? (MWOffliner or user?) "it" should do what when used with what command?

**Comment by kelson42** [comment]
@automactic `--useCache` should work like you have described.
